src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java

```diff
@@ -0,0 +1,68 @@
+package md.miller1995.Dealership.repositories;
+
+import md.miller1995.Dealership.models.entities.CarEntity;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+@DataJpaTest
+class CarRepositoryTest {
+
+    @Autowired
+    private CarRepository carRepository;
+    CarEntity carEntity;
+
+    @BeforeEach
+    void setUp() {
+
+        //given
+        carEntity = new CarEntity(
+                1L,
+                "Mercedes",
+                "A Class",
+                "134334VCNY66",
+                LocalDate.now(),
+                "blue",
+                2999
+        );
+
+        carRepository.save(carEntity);
+    }
+
+    @AfterEach
+    void tearDown() {
```

Comment:
When we are working with the repository layer, we want to create needed resources for each test and have a teardown method after all the test methods were executed (not after each test). In this way, we can save some resources.

Let's consider 2 examples, as a flow:
First approach:
```
1. Create a CarEntity before each test
2. Run test 1
3. Erase all the database records after each test

1. Create a CarEntity before each test
2. Run test 2
3. Erase all the database records after each test
```

Second approach:
```
1. Create a CarEntity inside the test in such a way you need it (maybe with some custom fields)
2. Run test 1

1. Create a CarEntity inside the test in such a way you need it (maybe with some custom fields)
2. Run test 2
...
3. Erase all the database records after all the tests were executed
```

Reply:
understood, but until I have only 2 tests for repository layer, I will use the first approach and after that I will update my code